mifosng-android/src/main/java/com/mifos/mifosxdroid/online/generatecollectionsheet/GenerateCollectionSheetPresenter.java

```diff
@@ -155,15 +222,356 @@ public void onCompleted() {
+
+                    @Override
+                    public void onNext(List<CenterDetail> centerDetails) {
+                        getMvpView().showProgressbar(false);
+                        getMvpView().onCenterLoadSuccess(centerDetails);
+                    }
+                }));
+    }
+
+    /**
+     * Method to fetch Productive Collection Sheet
+     */
+    public void loadProductiveCollectionSheet(int centerId,
+                                              CollectionSheetRequestPayload payload) {
+        checkViewAttached();
+        getMvpView().showProgressbar(true);
+        mSubscription.add(collectionDataManager.fetchProductiveCollectionSheet(centerId, payload)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Subscriber<CollectionSheetResponse>() {
+                    @Override
+                    public void onCompleted() {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        getMvpView().showProgressbar(false);
+                        try {
+                            if (e instanceof HttpException) {
+                                String errorMessage = ((HttpException) e).response().errorBody()
+                                        .string();
+                                getMvpView().showError(MFErrorParser.parseError(errorMessage)
+                                        .getErrors().get(0).getDefaultUserMessage());
+                            }
+                        } catch (Throwable throwable) {
+                            RxJavaPlugins.getInstance().getErrorHandler().handleError(e);
+                        }
+                    }
+
+                    @Override
+                    public void onNext(CollectionSheetResponse sheet) {
+                        getMvpView().showProgressbar(false);
+                        getMvpView().showProductive(sheet);
+                    }
+                }));
+    }
+
+    public void loadCollectionSheet(int groupId,
+                                              CollectionSheetRequestPayload payload) {
+        checkViewAttached();
+        getMvpView().showProgressbar(true);
+        mSubscription.add(collectionDataManager.fetchCollectionSheet(groupId, payload)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Subscriber<CollectionSheetResponse>() {
+                    @Override
+                    public void onCompleted() {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        getMvpView().showProgressbar(false);
+                        try {
+                            if (e instanceof HttpException) {
+                                String errorMessage = ((HttpException) e).response().errorBody()
```

Comment:
use this utils method to show error https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/utils/MFErrorParser.java#L25